src/Classes/PassiveTreeView.lua

```diff
@@ -271,42 +271,42 @@ function PassiveTreeViewClass:Draw(build, viewPort, inputEvents)
 	
 	local hotkeyPressed = IsKeyDown("1") or IsKeyDown("I") or IsKeyDown("2") or IsKeyDown("S") or IsKeyDown("3") or IsKeyDown("D")
 
-	-- Helper function to determine if global node allocation should be blocked
-	local function shouldBlockGlobalNodeAllocation(node)
-		local isGlobalNode = node.type == "Keystone" or node.type == "Socket" or node.containJewelSocket
-
-		if not isGlobalNode or node.alloc or not node.path then
+	-- Unified helper function to determine if node allocation should be blocked
+	local function shouldBlockNodeAllocation(node)
+		if not node.path or node.alloc then
 			return false
 		end
 
-		local weaponSetMode = spec.allocMode > 0
-		local connectedToWeaponSetNodes = self:IsConnectedToWeaponSetNodes(node)
-
-		-- Only allow allocation from main tree AND node must not be connected to weapon set nodes
-		local shouldBlock = weaponSetMode or connectedToWeaponSetNodes
+		local isGlobalNode = node.type == "Keystone" or node.type == "Socket" or node.containJewelSocket
 
-		return shouldBlock
+		if isGlobalNode then
+			local weaponSetMode = spec.allocMode > 0
+			local connectedToWeaponSetNodes = self:IsConnectedToWeaponSetNodes(node)
+			return weaponSetMode or connectedToWeaponSetNodes
+		else
+			return not self:PathCrossesWeaponSets(node, spec)
```

Comment:
This function doesn't exist, so I get an error when pathing past a jewel socket:
<img width="974" height="339" alt="image" src="https://github.com/user-attachments/assets/8f5f42df-a4f2-408f-923e-caa183186440" />